error_test.go

```diff
@@ -0,0 +1,37 @@
+package midtrans
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+)
+
+func TestErrorStruct(t *testing.T) {
+	err := &Error{
+		Message:        "Error Test Message",
+		StatusCode:     200,
+		RawError:       errors.New("TEST FROM GO ERROR"),
+		RawApiResponse: nil,
+	}
+	assert.Error(t, err)
+	assert.Equal(t, `Error Test Message`, err.GetMessage())
+	assert.Equal(t, 200, err.GetStatusCode())
+	assert.Equal(t, `TEST FROM GO ERROR`, err.Error())
+}
+
+func TestErrorResponse(t *testing.T) {
+	serverKey := "dummy"
+	c := GetHttpClient(Environment)
+	jsonReq, _ := json.Marshal("{\"transaction_details\": {\"order_id\": \"TEST-1648108994111\", \"gross_amount\": 10000}}")
+	err := c.Call(http.MethodPost, "https://app.midtrans.com/snap/v1/transactions", &serverKey, nil, bytes.NewBuffer(jsonReq), nil)
+
+	var midErr = err.Unwrap()
+	assert.True(t, true, errors.Is(err, midErr))
```

Comment:
from what I understand, error.Is is used to compare if the exact error object-instance (or wrapped error) `is equal`. It is tricky to test, I think you should just test it by comparing against itself e.g:
```golang
assert.True(t, true, errors.Is(err, err))
// or if it have .RawError:
// assert.True(t, true, errors.Is(err, err.RawError))
```

Reply:
Should also assert error.As, because it compares if the error object's type is equal to the defined error type.
Check [my code to see how to use error.As](https://replit.com/@raizerde/errorUnwrapAsIs#main.go).